code/modules/mob/living/simple_animal/hostile/plague_rat.dm

```diff
@@ -0,0 +1,88 @@
+/mob/living/simple_animal/hostile/plaguerat
+	name = "rat"
+	desc = "It's a nasty, ugly, evil, disease-ridden rodent with anger issues."
+	icon_state = "mouse_gray"
+	icon_living = "mouse_gray"
+	icon_dead = "mouse_gray_dead"
+	speak = list("Skree!","SKREEE!","Squeak?")
+	speak_emote = list("squeaks")
+	emote_hear = list("Hisses.")
+	emote_see = list("runs in a circle.", "stands on its hind legs.")
+	melee_damage_lower = 5 //stronk
+	melee_damage_upper = 5
+	obj_damage = 5
+	speak_chance = 1
+	turns_per_move = 5
+	see_in_dark = 6
+	maxHealth = 20
+	health = 20
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab/mouse = 1)
+	density = FALSE
+	ventcrawler = VENTCRAWLER_ALWAYS
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB
+	mob_size = MOB_SIZE_TINY
+	mob_biotypes = list(MOB_ORGANIC,MOB_BEAST)
+	faction = list("rat")
+	var/playstyle_string = "<span class='big bold'>You are a plague rat,</span></b> \
+							You need to infect people with plague, buy getting plague cells in their bloodstream. \
+							You can do it by licking their food, or just biting them. \
+							Also you can bite dead bodies to slightly heal yourself. \
+							But remember that you are very fragile - after all, you are a small rat.</b>"
+
+
+/mob/living/simple_animal/hostile/plaguerat/AttackingTarget()
+	..()
+	var/mob/living/L = target
+	if(isliving(target))
+		if(target.reagents)
+			L.reagents.add_reagent(/datum/reagent/plaguebacteria, 3)
+	if(L.stat == DEAD)
+		src.visible_message(span_warning("[src] starts biting into [L]!"),span_notice("You start eating [L]..."))
+		if(do_mob(src, target, 3 SECONDS))
+			to_chat(src, span_notice("You finish licking [L]."))
+			heal_bodypart_damage(5)
+			L.adjustBruteLoss(10)
+
+
+
+	if (target.reagents && target.is_injectable(src, allowmobs = TRUE))
```

Comment:
```suggestion
	if (!isliving(target) && target.reagents && target.is_injectable(src, allowmobs = TRUE))
```

I think currently this proc will add 3u of plague bacteria on line 38 and then start the licking message, is that intentional?

Reply:
Licking should work only in food, if the target isn’t a piece of food it would lick it, if it is a human it would bite it injecting reagents, and if it is a dead human it would start eating him